src/vdom.rs

```diff
@@ -143,39 +286,73 @@ where
             }
             (&Node::Text(TextNode { .. }), Node::Element(ElementNode { .. })) => {
                 debug!("  replacing a text node with an element");
-                self.create(new);
+                self.create(registry, new);
                 self.change_list.emit_replace_with();
             }
             (&Node::Element(ElementNode { .. }), Node::Text(TextNode { .. })) => {
                 debug!("  replacing an element with a text node");
-                self.create(new);
+                self.create(registry, new);
                 self.change_list.emit_replace_with();
             }
             (
                 &Node::Element(ElementNode {
                     tag_name: new_tag_name,
+                    listeners: new_listeners,
                     attributes: new_attributes,
                     children: new_children,
                 }),
                 Node::Element(ElementNode {
                     tag_name: old_tag_name,
+                    listeners: old_listeners,
                     attributes: old_attributes,
                     children: old_children,
                 }),
             ) => {
                 debug!("  updating an element");
                 if new_tag_name != old_tag_name {
                     debug!("  different tag names; creating new element and replacing old element");
-                    self.create(new);
+                    self.create(registry, new);
                     self.change_list.emit_replace_with();
                     return;
                 }
+                self.diff_listeners(registry, old_listeners, new_listeners);
                 self.diff_attributes(old_attributes, new_attributes);
-                self.diff_children(old_children, new_children);
+                self.diff_children(registry, old_children, new_children);
             }
         }
     }
 
+    fn diff_listeners<'a>(
+        &self,
+        registry: &mut EventsRegistry,
+        old: &'a [Listener<'a>],
+        new: &'a [Listener<'a>],
+    ) {
+        debug!("  updating event listeners");
+
+        'outer1: for new_l in new {
+            unsafe {
```

Comment:
So I've ended up here sort of randomly without actually knowing much context for what's going on, but as a random stylistic note might be good to have a comment about the unsafety here (even if it just points elsewhere)

Reply:
Good call. The comment you were looking for is on the `EventsRegistry::add` method, FWIW.